### PR TITLE
Add .travis.yml and simple build script

### PR DESCRIPTION
**Pull request by eozd**
This PR adds a Travis CI config file and a simple build script to build the project and run the tests in verbose mode.

##### Notes
* Currently, we build the project on Travis CI using various versions of gcc and clang. In order to make the code compileable with clang, I had to change some type arguments in the code.
* Travis supports cmake up to version 3.9.2. So, I downgraded the cmake version to that.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+mkdir -p build
+cd build
+cmake "$@" ..
+make
+cd build
```
Comment (ktrianta) [inline build.sh]: The second ```cd build``` can be left out.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -12,3 +12,4 @@ set(SRC_FILES
     main.cpp)
 
 add_executable(main ${SRC_FILES})
+target_link_libraries(main m)
```
Comment (ktrianta) [inline src/CMakeLists.txt]: Could you explain what this ```m``` is?

Comment (eozd) [reply src/CMakeLists.txt]: Math dll libm. It was required for clang linkage, I believe.

File: build.sh (hunk 1 above)
Comment (eozd) [reply build.sh]: Right :+1:

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.9.2)
```
Comment (RajaSimha) [inline CMakeLists.txt]: Just curious, why was the CMake's version demoted?

Comment (eozd) [reply CMakeLists.txt]: 3.9.2 is the highest version supported by Travis.

**Pull request merged**